### PR TITLE
The size of spline coefficients needs size_t

### PR DESCRIPTION
**Pull request by ye-luo**
Without the fix, I got 2292.59 MB. After the fix, I got 39156.6 MB.

### CODE REVIEW

File: src/Drivers/check_spo.cpp [hunk 1]
```diff
@@ -162,8 +162,8 @@ int main(int argc, char **argv)
     tileSize              = (tileSize > 0) ? tileSize : norb;
     nTiles                = norb / tileSize;
 
-    const unsigned int SPO_coeff_size =
-        (nx + 3) * (ny + 3) * (nz + 3) * norb * sizeof(RealType);
+    const size_t SPO_coeff_size =
+        (size_t)norb * (nx + 3) * (ny + 3) * (nz + 3) * sizeof(RealType);
```
Comment (PDoakORNL) [inline src/Drivers/check_spo.cpp]: The size_t should be unecessary, do you have a compiler that requires it? The fewer unecessary casts the better.  Same for the second one.

**Review by PDoakORNL — CHANGES_REQUESTED** [review]

**Comment by PDoakORNL** [comment]
Also without to fix is there a run that exposes the bug, or is this just for cleanliness and if we end up on a platform with size_t != unsigned int

File: src/Drivers/check_spo.cpp (hunk 1 above)
Comment (ye-luo) [reply src/Drivers/check_spo.cpp]: Do you mean the ```(size_t)norb```? I was not sure that all compilers will do 64bit integer.

**Comment by ye-luo** [comment]
See above. Default is 32bit integer.

File: src/Drivers/check_spo.cpp (hunk 1 above)
Comment (PDoakORNL) [reply src/Drivers/check_spo.cpp]: Yes that is the one I mean my understanding of C++'s implicit conversion is all the int should be getting promoted to size_t if its the larger type.

Comment (ye-luo) [reply src/Drivers/check_spo.cpp]: Could you point me the standard? I'd like to have a read.

Comment (ye-luo) [reply src/Drivers/check_spo.cpp]: I checked without this cast, I got 6388.59MB. Still wrong. This cast is needed.
For my understanding of the implicit conversion, the computing is 32 bit and only the final assign promotes the type without the explicit casting.

Comment (PDoakORNL) [reply src/Drivers/check_spo.cpp]: Here's the cppreference
https://en.cppreference.com/w/cpp/language/operator_arithmetic#Conversions
I'm not sure which standards docs are key.

Comment (ye-luo) [reply src/Drivers/check_spo.cpp]: It seems consistent with what I saw. The left to right rule applies here. All my changes are necessary.

Comment (PDoakORNL) [reply src/Drivers/check_spo.cpp]: Ah I see, I think lisp has made me a little worse at C++

Comment (PDoakORNL) [reply src/Drivers/check_spo.cpp]: Prefer static_cast<size_t>() to (size_t) for compile time check?

Comment (ye-luo) [reply src/Drivers/check_spo.cpp]: Intel on my workstation gives all these numbers. Only when both changes applied I get the correct number.

Comment (PDoakORNL) [reply src/Drivers/check_spo.cpp]: You are correct sir left to right applies so a cast on the first value is necessay.

```
#include <limits>
#include <iostream>

int main()
{
  size_t sizeof_size = std::numeric_limits<int>::max() * std::numeric_limits<int>::max() * sizeof(int);
  std::cout << sizeof_size << '\n';
  unsigned int tu_size = (unsigned int)std::numeric_limits<int>::max() * std::numeric_limits<int>::max();
  std::cout << tu_size << '\n';
  size_t test_size = static_cast<size_t>(std::numeric_limits<int>::max()) * std::numeric_limits<int>::max();
  std::cout << test_size << '\n';
}
```

```
bash-4.4$ g++-7 test.cpp
test.cpp: In function 'int main()':
test.cpp:6:56: warning: integer overflow in expression [-Woverflow]
   size_t sizeof_size = std::numeric_limits<int>::max() * std::numeric_limits<int>::max() * sizeof(int);
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
bash-4.4$ ./a.out
4
1
4611686014132420609
bash-4.4$ 
```

**Comment by PDoakORNL** [comment]
Opinion on writing such casts like commit 7d5094c

**Review by PDoakORNL — APPROVED** [review]

**Pull request merged**